promql/parse.go

```diff
@@ -1086,15 +1086,21 @@ func (p *parser) checkType(node Node) (typ model.ValueType) {
 
 	case *Call:
 		nargs := len(n.Func.ArgTypes)
-		if na := nargs - n.Func.Variadic; n.Func.Variadic >= 0 && na > len(n.Args) {
-			p.errorf("expected at least %d argument(s) in call to %q, got %d", na, n.Func.Name, len(n.Args))
-		}
-		if n.Func.Variadic >= 0 && nargs < len(n.Args) {
-			p.errorf("expected at most %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
-		}
-		if n.Func.Variadic < 0 && nargs > len(n.Args)+1 {
-			p.errorf("expected at least %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
+		if n.Func.Variadic == 0 {
+			if nargs > len(n.Args) {
+				p.errorf("expected at least %d argument(s) in call to %q, got %d", nargs, n.Func.Name, len(n.Args))
```

Comment:
If there's no variable arguments shouldn't this be exactly rather than at least/most?